### PR TITLE
Add Cocotb test framework for testing sim.v files

### PR DESCRIPTION
**Pull request by elmsfu**
An approach for #59 and super simple example of how to start #9

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -4,3 +4,6 @@
 [submodule "third_party/prjxray-db"]
 	path = third_party/prjxray-db
 	url = https://github.com/SymbiFlow/prjxray-db.git
+[submodule "third_party/cocotb"]
```
Comment (mithro) [inline .gitmodules]: Does it make sense to just install cocotb like we install pytest?

File: artix7/primitives/slicel/Nlut/simtest/test_alut.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+from cocotb.binary import BinaryValue
+from cocotb.triggers import Timer
+from cocotb.result import TestFailure
+from cocotb.regression import TestFactory
+from cocotb.generators.bit import random_50_percent
+
+import random
+
+class LutModel:
+  def __init__(self, init, inputs):
```
Comment (mithro) [inline artix7/primitives/slicel/Nlut/simtest/test_alut.py]: 4 spaces please. (PEP8 preferred...)

File: vpr/muxes/logic/mux2/simtest/Makefile [hunk 3]
```diff
@@ -0,0 +1,25 @@
+
```
Comment (mithro) [inline vpr/muxes/logic/mux2/simtest/Makefile]: Is this generated by cocotb?

File: artix7/primitives/slicel/Nlut/simtest/test_alut.py (hunk 2 above)
Comment (elmsfu) [reply artix7/primitives/slicel/Nlut/simtest/test_alut.py]: Done

File: .gitmodules (hunk 1 above)
Comment (elmsfu) [reply .gitmodules]: So add it to a conda channel like yosys and vtr? This seems simpler to me.

File: vpr/muxes/logic/mux2/simtest/Makefile (hunk 3 above)
Comment (elmsfu) [reply vpr/muxes/logic/mux2/simtest/Makefile]: I used the template and just changed a few variables. We can certainly simplify it.

File: .gitmodules (hunk 1 above)
Comment (mithro) [reply .gitmodules]: Try `env/conda/bin/pip install cocotb` -- or `env/conda/bin/pip install <git url>` see https://pip.pypa.io/en/stable/reference/pip_install/#git

Comment (mithro) [reply .gitmodules]: See how I added yapf in #103 pull request.

**Comment by mithro** [comment]
```

SymbiFlow Arch Defs - Running simulation tests
------------------------------------------------ 
./artix7/primitives/slicel/Nlut/simtest
Makefile:12: /home/travis/build/SymbiFlow/symbiflow-arch-defs/env/conda/lib/python3.6/site-packages/makefiles/Makefile.sim: No such file or directory
make[1]: *** No rule to make target '/home/travis/build/SymbiFlow/symbiflow-arch-defs/env/conda/lib/python3.6/site-packages/makefiles/Makefile.sim'.  Stop.
grep: ./artix7/primitives/slicel/Nlut/simtest/results.xml: No such file or directory
/bin/bash: line 0: [: ==: unary operator expected
./vpr/muxes/logic/mux2/simtest
Makefile:11: /home/travis/build/SymbiFlow/symbiflow-arch-defs/env/conda/lib/python3.6/site-packages/makefiles/Makefile.sim: No such file or directory
make[1]: *** No rule to make target '/home/travis/build/SymbiFlow/symbiflow-arch-defs/env/conda/lib/python3.6/site-packages/makefiles/Makefile.sim'.  Stop.
grep: ./vpr/muxes/logic/mux2/simtest/results.xml: No such file or directory
/bin/bash: line 0: [: ==: unary operator expected
make: *** [Makefile:46: simtest] Error 2
```

File: pylintrc [hunk 4]
```diff
@@ -0,0 +1,6 @@
+[MESSAGES CONTROL]
+enable=bad-indentation,eval-used,invalid-name,missing-docstring,old-style-class,protected-access,unused-import
+
+[REPORTS]
+output-format=colorized
+reports=no
```
Comment (mithro) [inline pylintrc]: Do you know if this matches yafp?

File: vpr/muxes/logic/mux2/simtest/Makefile (hunk 3 above)
Comment (mithro) [reply vpr/muxes/logic/mux2/simtest/Makefile]: Looks like this is simplified now.

**Comment by elmsfu** [comment]
I'm wondering if we should go back to adding cocotb as a submodule. The pip package is not officially supported and was essentially a one off. See: https://github.com/potentialventures/cocotb/issues/440

**Comment by mithro** [comment]
I'm going to merge this for now, I'll open issues around the submodule and cocotb makefiles.

**Comment by mithro** [comment]
Issue #111 and #112 created.

**Pull request merged**